parser/openfoodfacts_taxonomy_parser/parser/taxonomy_parser.py

```diff
@@ -438,6 +444,54 @@ def _get_valid_child_links(self, entry_nodes: list[NodeData], raw_child_links: l
 
         return valid_child_links
 
+    def _remove_duplicate_child_links(self, child_links: list[ChildLink]) -> list[ChildLink]:
+        """Remove duplicate child links (i.e child links with the same parent_id and id)"""
+        unique_child_links = []
+        children_to_parents = collections.defaultdict(set)
+        for child_link in child_links:
+            parent_id, child_id = child_link["parent_id"], child_link["id"]
+            if parent_id not in children_to_parents[child_id]:
+                children_to_parents[child_id].add(parent_id)
+                unique_child_links.append(child_link)
+        return unique_child_links
```

Comment:
FYI another way to do it:

```python
# classify by edges, overwriting if there is more than one value
edges_to_link = {
    (child_link["parent_id"], child_link["id"]): child_link
    for child_link in child_links
}
# return links
return list(values(edges_to_link))
    
```

Reply:
very aesthetic! 
However I'm not sure that overwriting is a good thing here because it could increase the diff. Indeed child links have a `position` property to keep the parent order at unparsing, so I think it's better to get the order of the parents that are common with other duplicates from the first node, because the new parents of the other duplicates will be added to the first node of the file during unparsing. This way there will only be new lines added to the parents of the first node, and no swap between the previous parents of the first node. I hope I'm clear

Reply:
No problem, it was just for fun ;-)